scripts.py

```diff
@@ -153,20 +153,38 @@ class IdentifierInputScript(Script):
     """A script that takes identifiers as command line inputs."""
 
     @classmethod
-    def parse_command_line(cls, _db=None, cmd_args=None, *args, **kwargs):
+    def read_stdin_lines(self, stdin):
+        """Read lines from a (possibly mocked, possibly empty) standard input."""
+        if stdin is not sys.stdin or not os.isatty(0):
+            # A file has been redirected into standard input. Grab its
```

Comment:
The os.isatty(0) part of this can't be automatically tested but I have tested it manually.